posix.mak

```diff
@@ -279,7 +284,14 @@ install :
 	$(MAKE) -f $(MAKEFILE) OS=$(OS) MODEL=$(MODEL) BUILD=release INSTALL_DIR=$(INSTALL_DIR) \
 		DMD=$(DMD) install2
 
+.PHONY : unittest
+ifeq (1,$(BUILD_WAS_SPECIFIED))
 unittest : $(addsuffix .d,$(addprefix unittest/,$(D_MODULES)))
+else
+unittest : unittest-debug unittest-release
+unittest-%:
+	$(MAKE) -f $(MAKEFILE) $(MAKEFLAGS) unittest OS=$(OS) MODEL=$(MODEL) DMD=$(DMD) BUILD=$*
```

Comment:
This broke my setup, I have a GNUmakefile that just forwards everything to posix.mak.

```
%:
	$(MAKE) DMD=$(DMD) -f posix.mak $*
```
Now this add additional nonsense arguments through MAKEFLAGS.

```
make DMD=../dmd/src/dmd -f posix.mak unittest
make[1]: Entering directory '/home/dawg/Code/D/DPL/phobos'
make -f posix.mak w -- DMD=../dmd/src/dmd unittest OS=linux MODEL=64 DMD=../dmd/src/dmd BUILD=debug
make[2]: Entering directory '/home/dawg/Code/D/DPL/phobos'
make[2]: *** No rule to make target 'w'.  Stop.
```

That's because the parent GNUmakefile (mine) passes `-w, --print-directory       Print the current directory.`

You don't need to pass MAKEFLAGS, it's already done for you.
If you still wanted to pass them you'd need `MAKEFLAGS=$(MAKEFLAGS)`.

The sub-make will communicate with the parent make over a jobs server so that no more than `-j N` rules run in parallel.

http://www.gnu.org/software/make/manual/make.html#MAKE-Variable
https://www.gnu.org/software/make/manual/html_node/Options_002fRecursion.html